### PR TITLE
Document all Flank properties in README

### LINKED ISSUE #458: Add smartFlankGcsPath to readme, flank.yml, and flank.ios.yml

### LINKED ISSUE #461: Audit readme
Make sure all supported options are listed

--results-dir is missing

---

```yaml
# test-targets-always-run:
  #   - a/testGrantPermissions

  # # test targets - a list of tests to run. omit to run all tests.
  #
  # test-targets:
  #   - b/testBasicSelection
```

clarify that the format on iOS is className/testName

### LINKED ISSUE #463: Improve xctestrun documentation
my .zip file and .xctestrun are in different folders. 

Seems like flank is trying to find xctest in a wrong directory (related where custom .xctestrun is). 

```
java -version
java version "11.0.2" 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.2+7-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.2+7-LTS, mixed mode)
```

```
java -jar ./flank.jar -v
v4.2.0
5148c339e697778a3714bfb1aa6d4a9a120c5164
```

```
xxx@xxx-27:~/ios/test_runner$ ls -la
total 49168
drwxr-xr-x   4 x  staff   128B Jan 17 09:37 ./
drwxr-xr-x@ 53 x  staff   1.7K Jan 17 09:03 ../
-rw-r--r--   1 x  staff   1.5K Jan 17 09:37 flank.ios.yml
-rw-r--r--   1 x  staff    24M Jan 11 23:27 flank.jar
x@xxx:~/ios/test_runner$ java -jar ./flank.jar firebase test ios run
IosArgs
    gcloud:
      results-bucket: test-lab-xxx
      results-dir: null
      record-video: true
      timeout: 30m
      async: false
      project: xxx
      results-history-name: null
      # iOS gcloud
      test: /Users/x/MyTests.zip
      xctestrun-file: /Users/x/Desktop/EmailSignupMethodTest_XXXUITests_iphoneos12.1-arm64e.xctestrun
      xcode-version: 10.1
      device:
        - model: iphonex
          version: 12.0
          locale: en_US
          orientation: portrait
      flaky-test-attempts: 3

    flank:
      testShards: -1
      repeatTests: 1
      smartFlankGcsPath: 
      test-targets-always-run:

      files-to-download:
        - .*\.png$
      # iOS flank
      test-targets:

RunTests
  Uploading MyTests.zip ..
Found xctest: /Users/x/Desktop/Debug-iphoneos/xxxUITests-Runner.app/PlugIns/xxxUITests.xctest
Exception in thread "main" picocli.CommandLine$ExecutionException: Error while running command (ftl.cli.firebase.test.ios.IosRunCommand@2d3ef181): java.lang.RuntimeException: File /Users/x/Desktop/Debug-iphoneos/xxxUITests-Runner.app/PlugIns/xxxUITests.xctest/xxxUITests does not exist!
	at picocli.CommandLine.execute(CommandLine.java:1168)
	at picocli.CommandLine.access$800(CommandLine.java:141)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1367)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:1335)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at picocli.CommandLine.run(CommandLine.java:1974)
	at picocli.CommandLine.run(CommandLine.java:1920)
	at ftl.Main$Companion.main(Main.kt:48)
	at ftl.Main.main(Main.kt)
Caused by: java.lang.RuntimeException: File /Users/x/Desktop/Debug-iphoneos/xxxUITests-Runner.app/PlugIns/xxxUITests.xctest/xxxUITests does not exist!
	at ftl.ios.Parse.validateFile(Parse.kt:22)
	at ftl.ios.Parse.parseObjcTests$test_runner(Parse.kt:36)
	at ftl.ios.Xctestrun.testsForTarget(Xctestrun.kt:36)
	at ftl.ios.Xctestrun.findTestNames(Xctestrun.kt:94)
	at ftl.ios.Xctestrun.findTestNames(Xctestrun.kt:78)
	at ftl.args.IosArgs$testShardChunks$2.invoke(IosArgs.kt:63)
	at ftl.args.IosArgs$testShardChunks$2.invoke(IosArgs.kt:26)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at ftl.args.IosArgs.getTestShardChunks(IosArgs.kt)
	at ftl.run.IosTestRunner$runTests$2.invokeSuspend(IosTestRunner.kt:38)
	at ftl.run.IosTestRunner$runTests$2.invoke(IosTestRunner.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:186)
	at ftl.run.IosTestRunner.runTests(IosTestRunner.kt:23)
	at ftl.run.TestRunner.runTests(TestRunner.kt:53)
	at ftl.run.TestRunner.newRun(TestRunner.kt:350)
	at ftl.cli.firebase.test.ios.IosRunCommand$run$1.invokeSuspend(IosRunCommand.kt:32)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.kt:116)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:76)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:53)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at ftl.cli.firebase.test.ios.IosRunCommand.run(IosRunCommand.kt:31)
	at picocli.CommandLine.execute(CommandLine.java:1160)
	... 9 more
x@x:~/ios/test_runner$
```

### PR DESCRIPTION
**Pull request by bootstraponline**
Fix #461 
Fix #458 
Fix #463

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,42 +46,102 @@ Run `test_runner/flank.ios.yml` with flank to verify iOS execution is working.
...
+  # xcode-version: 10.1
+
+  ## A list of DIMENSION=VALUE pairs which specify a target device to test against.
+  ## This flag may be repeated to specify multiple devices.
+  ## The four device dimensions are: model, version, locale, and orientation.
+  # device:
+  #  - model: iphone8
+  #   version: 12.0
+  #   locale: en
+  #   orientation: portrait
```
Comment (Macarse) [inline README.md]: why don't you add a second one here, so everyone knows how to work with more than one device?

```
      - model: iphonex
        version: 12.0
        locale: es_ES
        orientation: landscape
```

File: README.md [hunk 2]
```diff
@@ -46,42 +46,102 @@ Run `test_runner/flank.ios.yml` with flank to verify iOS execution is working.
...
+  ## This Xcode version must be supported by all iOS versions selected in the test matrix.
+  # xcode-version: 10.1
+
+  ## A list of DIMENSION=VALUE pairs which specify a target device to test against.
+  ## This flag may be repeated to specify multiple devices.
+  ## The four device dimensions are: model, version, locale, and orientation.
+  # device:
+  #  - model: iphone8
+  #   version: 12.0
+  #   locale: en
```
Comment (Macarse) [inline README.md]: I am using `en_US` instead of `en` alone.

Comment (bootstraponline) [reply README.md]: both are valid

File: README.md (hunk 1 above)
Comment (bootstraponline) [reply README.md]: great idea, done :)

File: README.md (hunk 2 above)
Comment (bootstraponline) [reply README.md]: image
https://github.com/TestArmada/flank/blob/master/test_runner/src/test/kotlin/ftl/fixtures/ios_catalog.json

**Pull request merged**